src/requirements.cpp

```diff
@@ -338,9 +348,74 @@ void requirement_data::check_consistency()
     }
 }
 
+template <typename T>
+void print_nested( const T &to_print, std::stringstream &ss )
+{
+    ss << "\n[ ";
+    for( auto &p : to_print ) {
+        print_nested( p, ss );
+        ss << " ";
+    }
+    ss << " ]\n";
+}
+
+template <typename T, typename Getter>
+void inline_requirements( std::vector< std::vector<T> > &list, Getter getter )
+{
+    std::set<requirement_id> already_nested;
+    for( auto &vec : list ) {
+        // We always need to restart from the beginning in case of vector relocation
+        while( true ) {
+            auto iter = std::find_if( vec.begin(), vec.end(), []( const T &req ) {
+                return req.requirement;
+            } );
+            if( iter == vec.end() ) {
+                break;
+            }
+
+            const auto req_id = requirement_id( iter->type );
+            if( !req_id.is_valid() ) {
+                debugmsg( "Tried to inline unknown requirement %s", req_id.c_str() );
+                return;
+            }
+
+            if( already_nested.count( req_id ) > 0 ) {
+                debugmsg( "Tried to inline requirement %s which was inlined before in the same pass (infinite loop?)",
+                          req_id.c_str() );
+                return;
+            }
+
+            already_nested.insert( req_id );
+            const auto &req = req_id.obj();
+            if( !req.get_qualities().empty() ) {
+                debugmsg( "Tried to inline requirement %s with qualities set (not supported)", req_id.c_str() );
+                return;
+            }
+
+            // The inlined requirement must have ONLY the type of component we are inlining
+            // That is, tools or components, not both (nor neither)
+            // Also, it must only offer alternatives, not more than one component "family" at a time
+            // @todo Remove the requirement to separate tools and components
+            // @todo Remove the requirement to have only one component "family" per inlined requirement
+            if( req.get_components().size() + req.get_tools().size() != 1 ) {
+                debugmsg( "Tried to inline requirement %s which has more than one set of elements", req_id.c_str() );
```

Comment:
@Coolthulhu Do you remember why this check is necessary?

I've removed this `if` block to make following requirements possible:

```json
  {
    "id": "soldering_standard",
    "type": "requirement",
    "//": "Soldering metal items",
    "tools": [ [ [ "soldering_iron", 1 ], [ "toolset", 1 ] ] ],
    "components": [ [ [ "solder_wire", 1 ] ] ]
  },
```

And it seems to be working properly - both solder and soldering tools are required - and they are spent properly.

![image](https://user-images.githubusercontent.com/16213433/42231955-b0c3ddae-7ef5-11e8-883c-d6924bb2b38f.png)

Reply:
Thanks. So it should work for simple requirements requiring set of tools/qulaities and list of components (like I did with sewing and soldering in #24180), but for more complex use case I will need to update the code.